FXyz-Core/src/main/java/org/fxyz3d/shapes/primitives/SegmentedDomeMesh.java

```diff
@@ -0,0 +1,231 @@
+import javafx.beans.property.SimpleDoubleProperty;
+import javafx.beans.property.SimpleIntegerProperty;
+import javafx.geometry.Point3D;
+import javafx.scene.DepthTest;
+import javafx.scene.shape.TriangleMesh;
+
+/**
+ *
+ * @author Sean
+ */
+public class SegmentedDomeMesh extends TexturedMesh {
+    /*
+        Field vars
+    */
+    private static final double DEFAULT_RADIUS = 50.0D;
+    private static final double DEFAULT_PHIMIN = Math.toRadians(0);
+    private static final double DEFAULT_PHIMAX = Math.toRadians(360);
+    private static final double DEFAULT_THETAMIN = Math.toRadians(0);
+    private static final double DEFAULT_THETAMAX = Math.toRadians(90);
+    private static final int DEFAULT_DIVISIONS = 20;    
+    
+    public SegmentedDomeMesh() {
+        this(DEFAULT_RADIUS, DEFAULT_PHIMIN, DEFAULT_PHIMAX, 
+            DEFAULT_THETAMIN, DEFAULT_THETAMAX, DEFAULT_DIVISIONS);
+    }
+    
+    /**
+     * @param radius radius of the sphere segment
+     * @param phimin The starting azimutal angle [rad], 0-2*pi.
+     * @param phimax The ending azimutal angle [rad], 0-2*pi, phimax &gt;
+     * phimin.
+     * @param thetamin The starting polar angle [rad], -pi/2-pi/2.
+     * @param thetamax The ending polar angle [rad], -pi/2-pi/2, thetamax &gt;
+     * thetamin.
+     * @param divisions The number of segments of curves approximations,
+     * granulariy &gt; 2.
+     */
+    public SegmentedDomeMesh(double radius,
+            double phimin, double phimax, 
+            double thetamin, double thetamax,
+            int divisions) {
+
+        setRadius(radius);
+        setPhimin(phimin);
+        setPhimax(phimax);
+        setThetamin(thetamin);
+        setThetamax(thetamax);
+        setDivisions(divisions);
+        setMesh(createSphereSegment(getRadius(), getPhimin(), getPhimax(), getThetamin(), getThetamax(), getDivisions()));
+    }
+    
+    private TriangleMesh createSphereSegment(double radius,
+            double phimin, double phimax, 
+            double thetamin, double thetamax,
+            int granularity) {
+    
+        setDepthTest(DepthTest.ENABLE);
+
+        mesh = new TriangleMesh();
+        // Fill Points
+        double phi = phimin;
+        double theta;
+
+        for (int i = 0; i < granularity + 1; i++) {
+            theta = thetamin;
+            for (int j = 0; j < granularity + 1; j++) {
+                Point3D p3D = new Point3D((float) (radius * Math.cos(theta) * Math.sin(phi)),
+                        (float) (radius * Math.cos(theta) * Math.cos(phi)),
+                        (float) (radius * Math.sin(theta)));
+                mesh.getPoints().addAll(new Float(p3D.getX()), new Float(p3D.getY()), new Float(p3D.getZ()));
+                theta += (thetamax - thetamin) / granularity;
+            }
+            phi += (phimax - phimin) / granularity;
+        }
+
+        //for now we'll just make an empty texCoordinate group
+        mesh.getTexCoords().addAll(0, 0);

```

Comment:
Important: regardless the value of the texture coordinates, having a TexturedMesh we need to use its objects like `ListVertices`, or `textureCoords`, otherwise when rendering the sample we'll have NPE or empty results. 
Again, you can check how the SegmentedSphere sets those lists.

Reply:
This has been implemented.